sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/training/FormTrainingAsyncClient.java

```diff
@@ -356,6 +472,88 @@ public FormRecognizerServiceVersion getServiceVersion() {
                 null));
     }
 
+    private Function<PollingContext<OperationResult>, Mono<CustomFormModelInfo>> fetchCopyModelResultOperation(
+        String modelId) {
+        return (pollingContext) -> {
+            try {
+                final UUID resultUid = UUID.fromString(pollingContext.getLatestResponse().getValue().getResultId());
+                Objects.requireNonNull(modelId, "'modelId' cannot be null.");
+                return service.getCustomModelCopyResultWithResponseAsync(UUID.fromString(modelId), resultUid)
+                    .map(modelSimpleResponse -> {
+                        CopyOperationResult copyOperationResult = modelSimpleResponse.getValue();
+                        return new CustomFormModelInfo(modelId,
+                            CustomFormModelStatus.fromString(copyOperationResult.getStatus().toString()),
+                            copyOperationResult.getCreatedDateTime(),
+                            copyOperationResult.getLastUpdatedDateTime());
+                    });
+            } catch (RuntimeException ex) {
+                return monoError(logger, ex);
+            }
+        };
+    }
+
+    private Function<PollingContext<OperationResult>, Mono<PollResponse<OperationResult>>>
+        createCopyPollOperation(String modelId) {
+        return (pollingContext) -> {
+            try {
+                PollResponse<OperationResult> operationResultPollResponse = pollingContext.getLatestResponse();
+                UUID targetId = UUID.fromString(operationResultPollResponse.getValue().getResultId());
+                return service.getCustomModelCopyResultAsync(UUID.fromString(modelId), targetId)
+                    .flatMap(modelSimpleResponse ->
+                        processCopyModelResponse(modelSimpleResponse, operationResultPollResponse));
+            } catch (HttpResponseException e) {
+                logger.logExceptionAsError(e);
+                return Mono.just(new PollResponse<>(LongRunningOperationStatus.FAILED, null));
+            }
+        };
+    }
+
+    private Function<PollingContext<OperationResult>, Mono<OperationResult>> getCopyActivationOperation(
+        String modelId, CopyAuthorization copyAuthorization) {
+        return (pollingContext) -> {
+            try {
+                Objects.requireNonNull(modelId, "'modelId' cannot be null.");
+                Objects.requireNonNull(copyAuthorization, "'copyAuthorization' cannot be null.");
```

Comment:
I think this is validation the parameters to beginCopyModel. If so, I think you should say "target" cannot be null